src/systems/joint_controller/JointController.cc

```diff
@@ -316,9 +326,19 @@ void JointController::PreUpdate(const UpdateInfo &_info,
 
   double error = jointVelComp->Data().at(0) - targetVel;
 
+  // If braking is disabled only apply force if the actual velocity is below
+  // the target - i.e. allow joint to freewheel.
+  bool applyForce = true;
+  if (this->dataPtr->disableBraking)
+  {
+      double sgn = math::signum(targetVel);
+      applyForce = sgn && sgn * error <= 0.0;
```

Comment:
I get a minor compile warning on my machine:

```sh
/home/iche/code/gz_i_ws_noble2/src/gz-sim/src/systems/joint_controller/JointController.cc:335:42: warning: comparing floating-point with ‘==’ or ‘!=’ is unsafe [-Wfloat-equal]
  335 |       applyForce = sgn && sgn * error <= 0.0;
      |                    ^~~
```
This should fix it

```suggestion
      applyForce = (sgn > 0) && sgn * error <= 0.0;
```

Reply:
We want to set `applyForce` false if `sgn == 0`, this is so that setting `targetVel = 0` will not brake when the joint has either a positive or negative velocity. So maybe `!(sgn == 0)` instead of just `(sgn)` - we are doing an 'is a float zero operation' - so maybe there's an existing util in `gz-math` we can use to make it robust?

Reply:
something like [math::equal](https://github.com/gazebosim/gz-math/blob/c35a71a2cafd910144090e13b27c8e93cd6507d1/include/gz/math/Helpers.hh#L383C15-L383C20), e.g. `!math::equal(sign, 0)` ?